qemu/tests/ksm_ksmtuned.py

```diff
@@ -31,41 +32,76 @@ def run(test, params, env):
+
+        def check_qemu_used_mem(qemu_pid, mem):
+            qemu_used_page = process.getoutput(get_qemu_used_mem % qemu_pid,
+                                               shell=True)
+            qemu_used_mem = float(qemu_used_page) * pagesize
+            if qemu_used_mem < mem * 0.95:
+                return False
+            return True
+
         params['mem'] = mem // 1024
         params['start_vm'] = 'yes'
         vm_name = params['main_vm']
         env_process.preprocess_vm(test, params, env, vm_name)
         vm = env.get_vm(vm_name)
-        vm.wait_for_login()
-        if stress and params["os_type"] == "linux":
+        session = vm.wait_for_login()
+        qemu_pid = vm.get_pid()
+        if params["os_type"] == "linux":
             params['stress_args'] = ('--cpu 4 --io 4 --vm 2 --vm-bytes %sM' %
                                      (int(params['mem']) // 2))
             stress_test = VMStress(vm, "stress", params)
             stress_test.load_stress_tool()
+        else:
+            install_path = params["install_path"]
+            test_install_cmd = 'dir "%s" | findstr /I heavyload' % install_path
+            heavyload_install()
+            heavyload_bin = r'"%s\heavyload.exe" ' % install_path
+            heavyload_options = ["/MEMORY 100", "/START"]
+            start_cmd = heavyload_bin + " ".join(heavyload_options)
+            stress_tool = BackgroundTest(session.cmd, (start_cmd,
+                                                       stress_timeout,
+                                                       stress_timeout))
+            stress_tool.start()
+            if not utils_misc.wait_for(stress_tool.is_alive, stress_timeout):
+                test.error("Failed to start heavyload process")
+        if not utils_misc.wait_for(lambda: check_qemu_used_mem(qemu_pid, mem),
+                                   stress_timeout, 10, 10):
+            test.error("QEMU used memory doesn't reach %sM in %ss"
+                       %(mem // 1024, stress_timeout))
         time.sleep(30)
-        qemu_pid = vm.get_pid()
-        qemu_used_page = utils_misc.normalize_data_size(process.getoutput(
-            params['cmd_get_qemu_used_mem'] % qemu_pid, shell=True) + 'K', 'B')
-        pagesize = utils_memory.getpagesize()
-        qemu_used_mem = int(float(qemu_used_page)) * pagesize
         free_mem_host = utils_memory.freememtotal()
         ksm_status = process.getoutput(params['cmd_check_ksm_status'])
         vm.destroy()
-        logging.info('The ksm threshold is %s, the memory allocated by qemu is'
-                     ' %s, and the total free memory on host is %s.',
-                     ksm_thres, qemu_used_mem, free_mem_host)
-        if free_mem_host >= ksm_thres:
+        logging.info('The ksm threshold is %sM, guest used memory is %sM, '
+                     'and the total free memory on host is %sM',
+                     ksm_thres // 1024, mem // 1024, free_mem_host // 1024)
```

Comment:
Two different description, `guest used memory` here and `qemu_used_mem` with function `check_qemu_used_mem`.   I think unique description is better.